common/utils/classnames.js

```diff
@@ -1,12 +1,19 @@
-export function withModifiers(className, modifiers = {}) {
+// @flow
+type SpacingCssProps = 'margin' | 'padding';
+type SpacingCssPropValues = 'top' | 'bottom' | 'left' | 'right';
+type SizeMap = { [string]: number };
+type SpacingProps = { [SpacingCssProps]: SpacingCssPropValues[] }
+type FontMap = { [string]: string }
+
+export function withModifiers(className: string, modifiers: { [string]: boolean } = {}) {
```

Comment:
We've removed all uses of `withModifiers()` (in favour of making `extraClasses` a string), so we can/should remove it altogether.

Reply:
Happy for someone too in another PR - this is just getting the typing right and trying to change as little runtime code as possible.